### PR TITLE
feat(api): New API module

### PR DESCRIPTION
**Pull request by ibgreen**
For #1525 

#### Background
- Gently start establishing a common WebGL / WebGPU API without breaking current WebGL code.
#### Change List
- New `@luma.gl/api` module with abstract classes for Buffer, Texture, Shader etc. 
- `@luma.gl/webgl` classes inherit from the API classes.

### CODE REVIEW

File: modules/api/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  "types": "src/index.ts",
+  "main": "dist/index.js",
+  "module": "dist/index.js",
+  "files": [
+    "src",
+    "dist",
+    "README.md"
+  ],
+  "sideEffects": false,
+  "browser": {
```
Comment (Pessimistress) [inline modules/api/package.json:28]: Remove? Given gl is not a dependency it should never be referenced.

File: modules/api/src/adapter/buffer.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+export default abstract class Buffer extends Resource<BufferProps> {
+  static MAP_READ = 0x01;
+  static MAP_WRITE = 0x02;
+
+  constructor(device: Device, props: BufferProps) {
+    super(device, props, DEFAULT_BUFFER_PROPS);
+  }
+
+  get [Symbol.toStringTag](): string {
+    return 'Buffer';
```
Comment (Pessimistress) [inline modules/api/src/adapter/buffer.ts:46]: Return id?

File: modules/api/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+luma.gl provides an abstract API for writing code that works with both WebGPU and WebGL.
+
+The API is modeled after the WebGPU API but is stream-lined to be less cumbersome to use.
+
+## Installing adapters
+
+The `@luma.gl/api` module is not usable on its own. A device adapter must be registered.
+
+```typescript
+import {luma} from '@luma.gl/api';
```
Comment (Pessimistress) [inline modules/api/docs/README.md:12]: This will be confusing down the road since we also have a global variable `luma` for log level etc.

**Review by Pessimistress — APPROVED** [review]

File: modules/api/src/adapter/buffer.ts (hunk 2 above)
Comment (ibgreen) [reply modules/api/src/adapter/buffer.ts:46]: - [toStringTag](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toStringTag) should just return a constant string. 
- We do have a `toString` in the resource base class that adds the id.

File: modules/api/docs/README.md (hunk 3 above)
Comment (ibgreen) [reply modules/api/docs/README.md:12]: Fair point. We have some time to decide what to call this global variable.

File: modules/api/package.json (hunk 1 above)
Comment (ibgreen) [reply modules/api/package.json:28]: Yes. Will update in next PR

**Pull request merged**